src/pybind/mgr/orchestrator_cli/module.py

```diff
@@ -235,11 +235,32 @@ def _mds_add(self, svc_arg):
         return self._add_stateless_svc("mds", spec)
 
     @_write_cli('orchestrator rgw add',
-                "name=svc_arg,type=CephString",
-                'Create an RGW service')
-    def _rgw_add(self, svc_arg):
+                'name=svc_arg,type=CephString,req=false',
+                'Create an RGW service. A complete <rgw_spec> can be provided'\
+                ' using <-i> to customize completelly the RGW service')
+    def _rgw_add(self, svc_arg=None, inbuf=None):
+        """
+        """
+        usage = """
+Usage:
+  ceph orchestrator rgw add -i <json_file>
+  ceph orchestrator rgw add <zone_name>
+        """
+
+        if inbuf:
+            try:
+                rgw_spec = orchestrator.RGWSpec.from_json(json.loads(inbuf))
```

Comment:
`from_json` is of little help here: the same behaviour can be achieved this way:

```suggestion
                rgw_spec = orchestrator.RGWSpec(**json.loads(inbuf))
```

However, I don't like any of them. The `from_json()` method is just a dummy method, which basically passes everything to the class initializer. With methods that expect to process user inputs, we should put extra care in them, as they are often attack vectors for hacking a system.